### PR TITLE
upgrade website to react-map-gl v5

### PR DESCRIPTION
**Pull request by Pessimistress**
The website currently uses react-map-gl v3, which depends on a mapbox version that is going to be deprecated.

This change upgrades react-map-gl to v5 and simplifies the demo setup.

We do lose the transition between demos by dropping the `<MapGL><DeckGL/></MapGL>` pattern.

### CODE REVIEW

File: examples/website/icon/app.js [hunk 1]
```diff
@@ -155,23 +155,20 @@ export class App extends Component {
...
-    const {viewState, controller = true, baseMap = true} = this.props;
+    const {mapStyle = 'mapbox://styles/mapbox/dark-v9'} = this.props;
 
     return (
       <DeckGL
         layers={this._renderLayers()}
         initialViewState={INITIAL_VIEW_STATE}
-        viewState={viewState}
-        controller={controller}
+        controller={{dragRotate: false}}
```
Comment (xintongxia) [inline examples/website/icon/app.js]: why this is needed?

File: website/src/components/demos/layer-demo-base.js [hunk 2]
```diff
@@ -107,7 +104,14 @@ export default function createLayerDemoClass(settings) {
 
       return (
         <div>
-          <DeckGL pickingRadius={5} viewState={viewState} layers={layers} />
+          <DeckGL pickingRadius={5}
```
Comment (xintongxia) [inline website/src/components/demos/layer-demo-base.js]: formatting not correct

File: website/src/components/demos/line-demo.js [hunk 3]
```diff
@@ -31,10 +31,6 @@ export default class LineDemo extends Component {
     };
   }
 
-  static get viewport() {
-    return INITIAL_VIEW_STATE;
-  }
-
   static get mapStyle() {
     return MAPBOX_STYLES.DARK;
```
Comment (xintongxia) [inline website/src/components/demos/line-demo.js]: Does this style have be defined in every demo?

Comment (Pessimistress) [reply website/src/components/demos/line-demo.js]: Yes.

File: examples/website/icon/app.js (hunk 1 above)
Comment (Pessimistress) [reply examples/website/icon/app.js]: This demo should only be viewed top-down.

**Pull request merged**